lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -116,33 +175,34 @@ def import_kymotrackgroup_from_csv(filename, kymo, channel, delimiter=";"):
     """
 
     # TODO: File format validation could use some improvement
-    try:
-        data = np.loadtxt(filename, delimiter=delimiter)
-    except ValueError:  # Could not convert to float
-        raise IOError("Invalid file format!")
+    data, pylake_version, csv_version = _read_txt(filename, delimiter)
 
-    if data.ndim != 2 or data.shape[0] <= 2:
+    mandatory_fields = ["time (pixels)", "coordinate (pixels)"]
+    if not all(f in data for f in mandatory_fields):
         raise IOError("Invalid file format!")
 
-    indices = data[:, 0]
-    time_idx = data[:, 1]
-    tracks = np.unique(indices)
+    for t in data["time (pixels)"]:
+        if np.any(np.floor(t) != t):
+            warnings.warn(
+                RuntimeWarning(
+                    "File contains non-integer time indices; round-off errors may have occurred "
+                    "when loading the data"
+                ),
+                stacklevel=2,
+            )
 
-    if np.any(np.floor(time_idx) != time_idx):
-        warnings.warn(
-            RuntimeWarning(
-                "File contains non-integer time indices; round-off errors may have occurred when "
-                "loading the data"
-            ),
-            stacklevel=2,
-        )
+    def create_track(time, coord, min_length=None):
+        if min_length is not None:
+            min_length = float(np.unique(min_length).squeeze())
+        return KymoTrack(time.astype(int), coord, kymo, channel, min_length)
 
-    return KymoTrackGroup(
-        [
-            KymoTrack(time_idx[indices == k].astype(int), data[indices == k, 2], kymo, channel)
-            for k in tracks
-        ]
-    )
+    min_length_field = "minimum_length (-)"
+    fields = mandatory_fields
+    if min_length_field in data:
+        fields.append(min_length_field)
```

Comment:
the copying to `fields` isn't doing much since youre just copying the reference (`mandatory_fields` is updated also in the `if` block). I'd be fine with just sticking with the same named list, I think it's clear enough.

```suggestion
    if (min_length_field := "minimum_length (-)") in data:
        mandatory_fields.append(min_length_field)
```

![image](https://github.com/lumicks/pylake/assets/61475504/9fc5b216-d1fe-4e11-8bd4-90bab28eb662)

Reply:
👍